src/pages/card/edit/model.ts

```diff
@@ -0,0 +1,28 @@
+import { StartParams } from '@box/lib/page-routing';
+import { attach, createEvent, merge, sample } from 'effector-root';
+import { cardActionsModel } from '@box/features/card/actions';
+import { cardModel } from '@box/entities/card';
+import { historyPush } from '@box/entities/navigation';
+
+import { paths } from '../../paths';
+
+export const getCardByIdFx = attach({ effect: cardModel.getCardByIdFx });
+export const pageLoaded = createEvent<StartParams>();
+
+// Подгружаем данные после монтирования страницы
+sample({
+  source: pageLoaded,
+  fn: ({ params }) => params.cardId,
+  target: getCardByIdFx,
+});
+
+// Возвращаем на страницу карточки после сохранения/отмены изменений
+sample({
+  clock: merge([
+    cardActionsModel.submitChangesFx.done,
+    cardActionsModel.resetChanges,
+  ]),
+  source: cardModel.$currentCardId,
+  fn: (cardId) => (cardId ? paths.card(cardId) : paths.home()),
+  target: historyPush,
+});
```

Comment:
nitpick: я бы разделил флоу на две части, одно для reset, другое для сохранения.

```suggestion
sample({
  clock: changesReset,
  target: historyPush.prepend(paths.home),
})

sample({
  source: $currentCardId,
  clock: changesSubmitFx.done,
  target: historyPush.prepend(path.card),
})
```

Reply:
Не-не, тут соль же в том, что `paths.home` это fallback-страница, в случае если вдруг внезапно cardId не будет задан к этому моменту (он же nullable у нас, не забываем)

Reply:
Тут именно смысл в том, чтобы разделить два сценария и сделать их наглядными и легкими для чтения.

в оригинальном варианте появляются условия. А это показатель, что более чем один сценарий вписан в выражение

Reply:
Тогда не понимаю реализацию разделения))

По сути весь сценарий это "ОСНОВНАЯ ЛОГИКА" + "FALLBACK":

```js
if (cardId)
    redirect_to_card_page(cardId)
else  // fallback
    redirect_to_home_page()
```

Можно было бы с таким успехом даже просто guard повесить какой-нить, который бы вообще не редиректил, если нет cardId (и возможно это даж лучше, я просто хз как в эффекторе это сделать)

Reply:
Ну вот не так. Ты смотришь с точки зрения ифов.
А надо с точки зрения цепочки действий.
Построй цепочку действий по прохождению сценария, и ты сразу заметишь что она разветвится на «фоллбэк».
Лучшая практика это разделять эти цепочки и в эффектор-модели

Reply:
> Лучшая практика это разделять эти цепочки и в эффектор-модели

Нуууу, допустим оке - разделим даж fallback логику на уровне модели

Но как предлагаешь ?)

> То что ты выше предложил, это
> - один сценарий "changesReset" + "paths.home"
> - а другой сценарий "changesSubmitFx" + "paths.card"

Reply:
@sergeysova Отпиши пож свой варик, иначе придется проигнорить)))

Reply:
Тож если и поправлю - то по другой задаче

Пока блокер (но лучше опиши пож свои предложения по пункту более конкретно пож)

https://linear.app/atomix/issue/BOX-167/dorabotat-modeli-do-best-praktis